packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/stackset-action.ts

```diff
@@ -0,0 +1,691 @@
+  readonly maxConcurrentPercentage?: number;
+
+  /**
+   * The AWS region the given Action resides in.
+   * Note that a cross-region Pipeline requires replication buckets to function correctly.
+   * You can provide their names with the {@link PipelineProps#crossRegionReplicationBuckets} property.
+   * If you don't, the CodePipeline Construct will create new Stacks in your CDK app containing those buckets,
+   * that you will need to `cdk deploy` before deploying the main, Pipeline-containing Stack.
+   *
+   * @default - the Action resides in the same region as the Pipeline
+   */
+  readonly region?: string;
+}
+
+/**
+ * Where Stack Instances will be created from the StackSet
+ */
+export abstract class StackInstances {
+  /**
+   * Create stack instances in a literal set of accounts or organizational units, and a set of regions
+   *
+   * Stack Instances will be created in every combination of region and account, or region and
+   * Organizational Units (OUs).
+   *
+   * If this is set of Organizational Units, you must have selected `StackSetDeploymentModel.organizations()`
+   * as deployment model.
+   */
+  public static fromList(targets: string[], regions: string[]): StackInstances {
+    if (targets.length === 0) {
+      throw new Error("'targets' may not be an empty list");
+    }
+
+    if (regions.length === 0) {
+      throw new Error("'regions' may not be an empty list");
+    }
+
+    return new class extends StackInstances {
+      public bind(_scope: Construct): StackInstancesBindResult {
+        return {
+          stackSetConfiguration: {
+            DeploymentTargets: targets.join(','),
+            Regions: regions.join(','),
+          },
+        };
+      }
+    }();
+  }
+
+  /**
+   * Create stack instances in a set of accounts or organizational units taken from the pipeline artifacts, and a set of regions
+   *
+   * Stack Instances will be created in every combination of region and account, or region and
+   * Organizational Units (OUs).
+   *
+   * If this is set of Organizational Units, you must have selected `StackSetDeploymentModel.organizations()`
+   * as deployment model.
+   */
+  public static fromArtifact(artifactPath: codepipeline.ArtifactPath, regions: string[]): StackInstances {
+    if (regions.length === 0) {
+      throw new Error("'regions' may not be an empty list");
+    }
+
+    return new class extends StackInstances {
+      public readonly artifactsReferenced?: codepipeline.Artifact[] | undefined = [artifactPath.artifact];
+      public bind(_scope: Construct): StackInstancesBindResult {
+        return {
+          stackSetConfiguration: {
+            DeploymentTargets: artifactPath.location,
+            Regions: regions.join(','),
+          },
+        };
+      }
+    }();
+  }
+
+  /**
+   * The artifacts referenced by the properties of this deployment target
+   *
+   * Does not need to be called by app builders.
+   */
+  readonly artifactsReferenced?: codepipeline.Artifact[];
```

Comment:
Any reason this isn't part of `StackInstancesBindResult`?

Also, can we make this `@internal`?

Reply:
> Any reason this isn't part of StackInstancesBindResult?

Yes! Because *someone* decided that the API of an Action needs to know the artifacts it's going to consume at `constructor` time, before `bind()` even happens 😛

Reply:
What moron made _that_ stupid decision??